crates/wasi-common/src/file.rs

```diff
@@ -317,3 +324,39 @@ pub enum Advice {
     DontNeed,
     NoReuse,
 }
+
+#[cfg(unix)]
+pub struct BorrowedAsFd<'a, T: AsFd>(&'a T);
```

Comment:
I think this type can be dropped as well? (a cursory look and I'm not seeing anywhere else it's used at least)